examples/cholesky_test.plv

```diff
@@ -0,0 +1,209 @@
+import cholesky;
+
+static diagonal {n}
+  (d :: double[n])
+  (out m :: double[n, n])
+  :: Void := (
+  for i in 0:n ->
+    for j in 0:n ->
+      if i == j then (m[i, j] <- d[i]) else (m[i, j] <- 0.0);
+);
+
+static tril {m, n}
+  (a :: double[m, n])
+  (out b :: double[m, n])
+  :: Void := (
+  for i in 0:m ->
+    for j in 0:n ->
+      if i < j then b[i, j] <- 0.0 else b[i, j] <- a[i, j];
+);
+
+static triu {m, n}
+  (a :: double[m, n])
+  (out b :: double[m, n])
+  :: Void := (
+  for i in 0:m ->
+    for j in 0:n ->
+      if i > j then b[i, j] <- 0.0 else b[i, j] <- a[i, j];
+);
```

Comment:
I just made this generic out of habit.  All the decompositions in this module require symmetric (and therefore square) matrices.  I'll try to use the built-in triangular matrix types.

I could also add the simple expression you wrote to the prelude.